src/main/java/org/junit/rules/TestRuleAdapter.java

```diff
@@ -0,0 +1,75 @@
+package org.junit.rules;
+
+import org.junit.runner.Description;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Method;
+
+/**
+ * A TestRuleAdapter allows to wrap {@link MethodRule} classes in
+ * {@link TestRule} classes in order to use them inside {@link RuleChain}.
+ * Assuming we have an existing method rule:
+ *
+ * <pre>
+ * class MyMethodRule implements MethodRule {
+ *     public Statement apply(final Statement base, FrameworkMethod method,
+ *                            Object target) {
+ *         return new Statement() {
+ *             &#064;Override
+ *             public void evaluate() throws Throwable {
+ *                 base.evaluate();
+ *             }
+ *         };
+ *     }
+ * }
+ * </pre>
+ *
+ * It can then be used in a {@link RuleChain} like this:
+ * <pre>
+ * class MyTest {
+ *     &#064;Rule
+ *     public final RuleChain chain = RuleChain
+ *             .outerRule(new TestRuleAdapter(new MyMethodRule(), this))
+ *             .around(new TestRuleAdapter(new MyMethodRule(), this));
+ *
+ *     ...
+ * }
+ * </pre>
+ *
+ * @since 4.12
+ */
+class TestRuleAdapter implements TestRule {
+    private final MethodRule rule;
+    private final Object testObject;
+
+    /**
+     * Creates a new instance wrapping the provided {@link MethodRule}.
+     *
+     * Needs an instance of the test class using this TestRuleAdapter or
+     * the surrounding {@link RuleChain} to be allow the wrapped
+     * {@link MethodRule} to interact with it.
+     *
+     * @param rule the MethodRule to wrap
+     * @param testObject instance of the test class
+     */
+    public TestRuleAdapter(MethodRule rule, Object testObject) {
+        this.rule = rule;
+        this.testObject = testObject;
+    }
+
+    public Statement apply(Statement base, Description description) {
+        try {
+            String methodName = description.getMethodName();
+            Method method = testObject.getClass().getDeclaredMethod(methodName);
```

Comment:
Marc is correct. Our options appear to be:

1. Update the Javadoc to mention this limitation, but otherwise merge this
2. Add a method to `Description` to get the `Method`, then merge this
3. Do not merge this

For option 2, I would prefer that we not add more static methods for creating `Description` instances. We could explore the work that was done a while back on creating builders for creating immutable `Description` instances (see https://github.com/junit-team/junit4/tree/immutable-description). I recently cleaned up the code in that branch, but it's a rather large set of changes, and the changes could break people.

I'm removing the "4.13" tag, since I don't want to block the next JUnit 4.x release on resolving this issue.